src/PhpSpreadsheet/Writer/Xls/Worksheet.php

```diff
@@ -441,7 +441,23 @@ public function close()
                     case DataType::TYPE_FORMULA:
                         $calculatedValue = $this->preCalculateFormulas ?
                             $cell->getCalculatedValue() : null;
-                        $this->writeFormula($row, $column, $cVal, $xfIndex, $calculatedValue);
+                        if (-3 == $this->writeFormula($row, $column, $cVal, $xfIndex, $calculatedValue)) {
+                            $calculatedValue = $cell->getCalculatedValue();
+                            $calctype = gettype($calculatedValue);
+                            switch ($calctype) {
+                                case 'integer':
+                                case 'double':
+                                    $this->writeNumber($row, $column, $calculatedValue, $xfIndex);
+
+                                    break;
+                                case 'string':
+                                    $this->writeString($row, $column, $calculatedValue, $xfIndex);
+
+                                    break;
+                                default:
```

Comment:
What is the result of the calculation is a bool?

Reply:
Newly added lines 459-461 address this issue. I have also added a test case for booleans, plus one for strings.